feat(ui-kit): CourseOutline component UI , storybook and tests are added.

manishRise11 opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: `ComponentName`
- Figma link or screenshot: [Figma Design](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=6687-25809&t=Odg9AWgp943rCt3H-4)
-![Image](https://github.com/user-attachments/assets/d9505f39-2b90-47d0-87a8-9120db6c2779)

---


## React Best Practices Checklist

- [ ] I have avoided **unnecessary re-renders** by using composition (`children` as props), whenever possible. This applies, in particular, to composite components (like lists of components, and the like).
- [ ] **I have designed my components as server-side whenever possible**—I have not used `useState`, `useEffect`, `useContext`, or any other React hook that works with "use client" without a good reason.
- [ ] **I have not put props in state**—in general, props should remain as props unless the component needs an `initialValue`, and judiciously manage its own independent state detached from the parent component.
    - If I have stored props in state, I have explained where exactly and justified it with a solid reason here below:
    
- [ ] **I have avoided using Context** in any critical component in terms of performance
    - If I have used Context, I have explained where exactly and justified it with a solid reason here below:


## General Checklist

- [ ] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [ ] My changes align with the **core models** that have been discussed and agreed upon.
- [ ] The component design has been compared to the **Figma design** to ensure consistency.
- [ ] The **component is fully responsive** and behaves as expected across different screen sizes.
- [ ] The component **supports keyboard web accessibility** (if applicable).
- [ ] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [ ] I have updated the **documentation** for the component, including usage examples and guidelines.
- [ ] I have **commented** my code, particularly in complex or non-obvious areas.
- [ ] The **component is properly exported** for use in other parts of the application.




---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #164

carlapbg reviewed (changes requested):
Thanks! Just one thing: The gap between the title and the badge should be smaller. When the title is not selected it looks good, but then if I selected it the gap is too big

![Screenshot 2025-05-01 at 21 20 04](https://github.com/user-attachments/assets/ee82b81a-3009-4ccf-ba52-f175585fa40c)
![Screenshot 2025-05-01 at 21 20 21](https://github.com/user-attachments/assets/79491625-a49d-4d7d-84e9-f07b49f9f334)

alebg commented on packages/ui-kit/lib/components/milestone.tsx:
```diff
@@ -0,0 +1,48 @@
+    completed,
+    locale
+}) => {
+    const disctionary = getDictionary(locale);
```
@manishRise11 please fix this typo

alebg commented on packages/ui-kit/lib/components/course-outline/lesson-link-item.tsx:
```diff
@@ -0,0 +1,102 @@
+    onClick,
+    locale
+}) => {
+    const disctionary = getDictionary(locale);
```
@manishRise11 please fix this typo

alebg commented on packages/ui-kit/stories/course-outline/course-outline-accordian.stories.tsx:
@manishRise11 please spell this "accordion"

alebg commented on packages/ui-kit/stories/course-outline/course-outline-accordian.stories.tsx:
```diff
@@ -0,0 +1,281 @@
+import { FC, useState } from 'react'; 
+
+const meta: Meta = {
+  title: 'Components/CourseOutline/CourseOutlineAccordian',
```
@manishRise11 same here, please spell this "accordion"

alebg commented on packages/ui-kit/stories/course-outline/course-outline-accordian.stories.tsx:
```diff
@@ -0,0 +1,281 @@
+  ]
+};
+
+// Define the CourseOutlineAccordian component
```
@manishRise11 for reference: https://www.merriam-webster.com/dictionary/accordian
The correct spelling is "accordion"

alebg reviewed (changes requested):
@manishRise11 thanks a lot. Please address my inline comments, and anything discussed over the calls

manishRise11 commented:
Hi @alebg sir , @carlapbg mam,

I've addressed all the requested changes:
- Fixed the typo from disctionary to dictionary in both milestone and lesson-link-item components.
- Corrected the spelling of "accordion" throughout the course-outline stories.
- Moved the milestone component and related stories to the course-outline folder as suggested.
- Reduced the gap between the title and badge in the milestone component.

Please review the latest changes when you have a moment. Let me know if there’s anything else to update!

alebg reviewed (dismissed):
@manishRise11 looks very good, thanks  a lot! Let's wait for @carlapbg 's final check and then we're good to merge this

carlapbg reviewed (approved):
Thank you @manishRise11 , looks good!
